nodes/imu_node.py

```diff
@@ -248,8 +248,8 @@ def write_and_check_config(serial_instance, calib_dict):
             #  when the chip is facing forward, in the sparkfun board, the chip is facing the left side
             # but Sparkfun the firmware interchanges x and y and changes the sign of y
             # so to get it to REP103 we need to swap X and Y again and make Y and Z negative
-            magMsg.magnetic_field.x = float(words[10]) * 1e-7
-            magMsg.magnetic_field.y = -float(words[9]) * 1e-7
+            magMsg.magnetic_field.x = float(words[9]) * 1e-7
+            magMsg.magnetic_field.y = -float(words[10]) * 1e-7
             magMsg.magnetic_field.z = -float(words[11]) * 1e-7
```

Comment:
Check line 178 form Sensors.ino
`    // Conversion from uT to mGauss.
    magnetom[0] = (10.0f*imu.calcMag(imu.my));
    magnetom[1] = (-10.0f*imu.calcMag(imu.mx));
    magnetom[2] = (10.0f*imu.calcMag(imu.mz));`

So after breaking my head around it again, This is how I think the frames should be:
            magMsg.magnetic_field.x = -float(words[9]) * 1e-7
￼             magMsg.magnetic_field.y = float(words[10]) * 1e-7
￼             magMsg.magnetic_field.z = -float(words[11]) * 1e-7

What do you think?

Reply:
![image](https://user-images.githubusercontent.com/4633760/68659933-1f4e0f80-0530-11ea-9f0c-48d0bddff527.png)
![image](https://user-images.githubusercontent.com/4633760/68659992-41479200-0530-11ea-8e00-03d2f4fd5ab4.png)
![image](https://user-images.githubusercontent.com/4633760/68660017-4e648100-0530-11ea-8107-1c412ceabb9c.png)